src/iter/mod.rs

```diff
@@ -215,6 +217,13 @@ pub trait ParallelIterator: Sized + Send {
         flat_map::new(self, map_op)
     }
 
+    /// Reduces a collection of items into one collection. If you have
+    /// a `Vec<Vec<i32>>` it can be used to turn it into a `Vec<i32>`.
+    fn flatten(self) -> Flatten<Self>
```

Comment:
Oh, and I forgot to mention, I think the documentation would be better if it spoke in terms of iterators, not collections.  Something like the `flat_map` docs without the mapping part. :)

Perhaps: "An adaptor that flattens iterable `Item`s into one large iterator."